platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/tasks/CachingTaskDependencyResolveContext.java

```diff
@@ -58,14 +59,27 @@ public CachingTaskDependencyResolveContext(Collection<? extends WorkDependencyRe
         this.walker = new CachingDirectedGraphWalker<>(new TaskGraphImpl(workResolvers));
     }
 
+    @Override
+    public String getName() {
+        if (task != null) {
+            return "task dependencies";
+        } else {
+            return "dependencies";
+        }
+    }
+
     public Set<T> getDependencies(@Nullable Task task, Object dependencies) {
         Preconditions.checkState(this.task == null);
         this.task = task;
         try {
             walker.add(dependencies);
             return walker.findValues();
         } catch (Exception e) {
-            throw new TaskDependencyResolveException(format("Could not determine the dependencies of %s.", task), e);
+            if (task != null) {
+                throw new TaskDependencyResolveException(format("Could not determine the dependencies of %s.", task), e);
+            } else {
+                throw UncheckedException.throwAsUncheckedException(e);
```

Comment:
I still think we should continue to throw a `TaskDependencyResolveException` here.

AFAIK task dependencies are called "task" dependencies not because they are the dependencies of a task, but because we are collecting all tasks that need to be run as dependencies to _something_. 

Though I am not 100% sure here, so someone from the @gradle/bt-execution team should be the final review on this PR IMO

Reply:
The reason I didn't do this is that `TDRE` would require a message, which would emit another line in the exception hierarchy.  Or we could use a blank message to make the `BuildExceptionReporter` skip it.  No strong feelings here, wrapping just seemed unnecessary.

Reply:
Resolved in conversation: leaving as is.